test/JDBC/expected/test_dynamic_local_vars.out

```diff
@@ -0,0 +1,1172 @@
+go
+~~START~~
+int
+1
+~~END~~
+
+
+DECLARE @a int
+set @a = 0
+select @a += ~@a
+select @a
+go
+~~START~~
+int
+-1
+~~END~~
+
+
+-- quoted identifiers
+declare @v varchar(20) = "ABC", @v2 varchar(20)="XYZ";
+select @v += "a""b''c'd", @v2 += "x""y''z";
+select @v, @v2
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "abc" does not exist)~~
+
+
+declare @v varchar(20) = "ABC", @v2 varchar(20)="XYZ";
+select @v += "a""b''c'd", @v2 += @v + "x""y''z";
+select @v, @v2
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "abc" does not exist)~~
+
+
+declare @v varchar(20) = "ABC", @v2 varchar(20)="XYZ";
+select @v += reverse("a""b''c'd"), @v2 += @v + "x""y''z";
+select @v, @v2
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "abc" does not exist)~~
+
+
+declare @v varchar(20) = "ABC", @v2 varchar(20)="XYZ";
+select @v += reverse("a""b''c'd"), @v2 += @v + reverse("x""y''z");
+select @v, @v2
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "abc" does not exist)~~
+
+
+declare @v varchar(20) = "ABC", @v2 varchar(20)="XYZ";
+select @v += reverse("a""b''c'd"), @v2 += REVERSE( @v + reverse("x""y''z"));
+select @v, @v2
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: column "abc" does not exist)~~
+
+
+declare @v varchar(20) = 'ABC', @v2 varchar(20)='XYZ';
+select @v += 'abc', @v2 += 'xyz';
+select @v, @v2
+GO
+~~START~~
+varchar#!#varchar
+ABCabc#!#XYZxyz
+~~END~~
+
+
+declare @a int = 1, @b int = 2;
+select @a = 2, @b = @a + 2
+select @a, @b
+GO
+~~START~~
+int#!#int
+2#!#4
+~~END~~
+
+
+declare @a int = 1, @b int = 2;
+select @a += 2, @b -= @a + 2
+select @a, @b
+GO
+~~START~~
+int#!#int
+3#!#-3
+~~END~~
+
+
+-- xml methods
+DECLARE @a bit = 1
+DECLARE @xml XML = '<artists> <artist name="John Doe"/> <artist name="Edward Poe"/> <artist name="Mark The Great"/> </artists>'
+SELECT @a |= @xml.exist('/artists/artist/@name')
+select @a
+GO
+~~START~~
+bit
+1
+~~END~~
+
+
+-- test all kind of udts
+create type udt from NCHAR
+go
+
+declare @a udt
+select @a = 'anc'
+select @a
+GO
+~~START~~
+nchar
+a
+~~END~~
+
+
+DROP type udt 
+GO
+
+create type varchar_max from varchar(max)
+GO
+
+DECLARE @a varchar_max
+SELECT @a = '12345678901234567890123456789012345';
+SELECT LEN(@a), DATALENGTH(@a)
+SELECT @a
+GO
+~~START~~
+int#!#int
+35#!#35
+~~END~~
+
+~~START~~
+varchar
+12345678901234567890123456789012345
+~~END~~
+
+
+DROP type varchar_max
+GO
+
+create type num_def from numeric
+GO
+
+declare @a numeric;
+declare @b num_def;
+SET @a=100.41;
+SET @b=200.82;
+SELECT @a, @b
+select @a+@b as r;
+GO
+~~START~~
+numeric#!#numeric
+100#!#201
+~~END~~
+
+~~START~~
+numeric
+301
+~~END~~
+
+
+drop type num_def
+GO
+
+/*
+ * UPDATE test
+ */
+create table local_var_tst (id int) 
+GO
+
+TRUNCATE table local_var_tst
+GO
+
+insert into local_var_tst values (1)
+insert into local_var_tst values (2)
+insert into local_var_tst values (6)
+GO
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+
+
+
+
+
+-- declare @i int 
```

Comment:
It will be fixed with separate work. https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/3028 https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/461/files